cloudsmith_cli/cli/validators.py

```diff
@@ -155,6 +155,17 @@ def validate_page_size(ctx, param, value):
     return value
 
 
+def validate_show_all(ctx, param, value):
+    """Ensure that --show-all is not used with --page (-p) or --page-size (-l)."""
+    if value:
+        # Check if either page or page_size parameters were provided, regardless of value
+        if any(param in ctx.params for param in ["page", "page_size"]):
+            raise click.UsageError(
+                "The --show-all option cannot be used with --page (-p) or --page-size (-l) options."
+            )
+    return value
+
+
```

Comment:
Can you add a test for this please?

Does the order in which the parameters are passed affect the validation?